### PR TITLE
[Fix] add type hints for bfp

### PR DESCRIPTION
**Pull request by JunyaoHu**
## Motivation

超级视客营 mmdet 基础任务，为part-9的 `bfp.py` 添加 type hints

## Modification

为 `bfp.py` 中相关类的函数添加 type hints

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/necks/bfp.py [hunk 1]
```diff
@@ -21,23 +25,23 @@ class BFP(BaseModule):
         in_channels (int): Number of input channels (feature maps of all levels
             should have the same channels).
         num_levels (int): Number of input feature levels.
-        conv_cfg (dict): The config dict for convolution layers.
-        norm_cfg (dict): The config dict for normalization layers.
         refine_level (int): Index of integration and refine level of BSF in
             multi-level features from bottom to top.
         refine_type (str): Type of the refine op, currently support
             [None, 'conv', 'non_local'].
+        conv_cfg (dict): The config dict for convolution layers.
```
Comment (zytx121) [inline mmdet/models/necks/bfp.py:32]:
```suggestion
        conv_cfg (:obj:`ConfigDict` or dict, optional): The config dict for convolution layers.
```

File: mmdet/models/necks/bfp.py [hunk 2]
```diff
@@ -21,23 +25,23 @@ class BFP(BaseModule):
...
+        norm_cfg (dict): The config dict for normalization layers.
```
Comment (zytx121) [inline mmdet/models/necks/bfp.py:33]:
```suggestion
        norm_cfg (:obj:`ConfigDict` or dict, optional): The config dict for normalization layers.
```

File: mmdet/models/necks/bfp.py [hunk 3]
```diff
@@ -21,23 +25,23 @@ class BFP(BaseModule):
...
-                 init_cfg=dict(
+                 in_channels: int,
+                 num_levels: int,
+                 refine_level: int = 2,
+                 refine_type: str = None,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: OptConfigType = None,
+                 init_cfg: OptMultiConfig = dict(
                      type='Xavier', layer='Conv2d', distribution='uniform')):
         super(BFP, self).__init__(init_cfg)
```
Comment (BIGWangYuDong) [inline mmdet/models/necks/bfp.py:46]:
```suggestion
        super().__init__(init_cfg=init_cfg)
```

File: mmdet/models/necks/bfp.py [hunk 4]
```diff
@@ -21,23 +25,23 @@ class BFP(BaseModule):
...
-                 norm_cfg=None,
-                 init_cfg=dict(
+                 in_channels: int,
+                 num_levels: int,
+                 refine_level: int = 2,
+                 refine_type: str = None,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: OptConfigType = None,
+                 init_cfg: OptMultiConfig = dict(
                      type='Xavier', layer='Conv2d', distribution='uniform')):
```
Comment (BIGWangYuDong) [inline mmdet/models/necks/bfp.py:45]: -> None

File: mmdet/models/necks/bfp.py [hunk 5]
```diff
@@ -67,7 +71,7 @@ def __init__(self,
                 conv_cfg=self.conv_cfg,
                 norm_cfg=self.norm_cfg)
 
-    def forward(self, inputs):
+    def forward(self, inputs=Tuple[Tensor]) -> Tuple[Tensor]:
```
Comment (zytx121) [inline mmdet/models/necks/bfp.py:74]:
```suggestion
    def forward(self, inputs: Tuple[Tensor]) -> Tuple[Tensor]:
```

File: mmdet/models/necks/bfp.py [hunk 6]
```diff
@@ -21,25 +25,26 @@ class BFP(BaseModule):
...
-                 norm_cfg=None,
-                 init_cfg=dict(
+                 in_channels: int,
+                 num_levels: int,
+                 refine_level: int = 2,
+                 refine_type: str = None,
+                 conv_cfg: OptConfigType = None,
+                 norm_cfg: OptConfigType = None,
+                 init_cfg: OptMultiConfig = dict(
                      type='Xavier', layer='Conv2d', distribution='uniform')):
```
Comment (BIGWangYuDong) [inline mmdet/models/necks/bfp.py:46]:
```suggestion
                     type='Xavier', layer='Conv2d', distribution='uniform')) -> None:
```

**Comment by JunyaoHu** [comment]
PTAL again. Thanks! By the way I try to solve CI problem, since it shows that 1 line should have less than 80 characters.

**Review by zytx121 — APPROVED** [review]
LGTM

**Pull request closed without merging**